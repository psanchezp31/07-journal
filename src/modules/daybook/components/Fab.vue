<template>
  <button class="btn btn-primary" @click="$emit('on:click')">
    <i class="fa fa-2x" :class="icon"></i>
    <!-- se pueden colocar dos class, pero uno con : y otro sin-->
  </button>
</template>

<script>
export default {
  data() {
    return {};
  },
  props: {
    icon: {
      type: String,
      default: "fa-plus",
      required: false,
    },
  },

};
</script>

<style lang="scss" scoped>
button {
  border-radius: 100%;
  bottom: 20px;
  height: 80px;
  position: fixed;
  right: 20px;
  width: 80px;
}
</style>
