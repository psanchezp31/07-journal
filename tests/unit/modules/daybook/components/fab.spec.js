import { shallowMount } from "@vue/test-utils";
import Fab from "@/modules/daybook/components/Fab.vue";

describe("Pruebas en el fab component", () => {
  test("debe de renderizar el componente correctamente", () => {
    const wrapper = shallowMount(Fab);
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("debe de mostrar el icono por defecto ", () => {
    const wrapper = shallowMount(Fab);
    // expect(wrapper.props().icon).toBe('fa-plus')
    expect(wrapper.find("i").classes("fa-plus")).toBeTruthy();
  });
  test("debe de mostar el Ã­cono por argumento : fa-circle ", () => {
    const wrapper = shallowMount(Fab, {
      props: {
        icon: "fa-circle",
      },
    });

    expect(wrapper.find("i").classes("fa-circle")).toBeTruthy();
  });
  test("debe de emitir el evento on:click cuando se hace click ", () => {
    const wrapper = shallowMount(Fab);
    wrapper.find("button").trigger("click");
    expect(wrapper.emitted('on:click')).toHaveLength(1);
  });
});
