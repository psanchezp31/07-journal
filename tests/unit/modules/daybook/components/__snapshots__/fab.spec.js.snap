// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el fab component debe de renderizar el componente correctamente 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i><!-- se pueden colocar dos class, pero uno con : y otro sin--></button>`;
